code/tasks/CleanupOrphanedRecordsTask.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+class CleanupOrphanedRecordsTask extends BuildTask {
+
+	public function run($request) {
+		$args = $request->getVar('args');
+		$dryRun = $args && in_array('--dry-run', $args);
+
+		$log = function($message) {
+			$message = sprintf('[%s] ', date('Y-m-d H:i:s')) . $message;
+			echo $message . PHP_EOL;
+		};
+
+		if (!Director::is_cli()) {
+			$log('This task must be run under the command line');
+			return;
+		}
+
+		if ($dryRun) {
+			$log('Running in dry-run mode. No data will be deleted');
+		}
+
+		$count = 0;
+
+		foreach (DNEnvironment::get() as $environment) {
+			$project = $environment->Project();
+			if (!$project || !$project->exists()) {
+				$log(sprintf(
+					'Environment (ID %s, Created: %s) is linked to a non-existent project. Deleting',
```

Comment:
Can you add the name of the environment to the output please?
`... is linked to a non-existent project 'Banana'`